autora/cycle/_simple_cycle.py

```diff
@@ -267,3 +389,97 @@ def _theorist_callback(theorist, data_in: _SimpleCycleData):
     def _monitor_callback(self, data: _SimpleCycleData):
         if self.monitor is not None:
             self.monitor(data)
+
+
+def _resolve_magic_params(params: Dict, magics: Mapping):
+    """
+    Resolve "magic values" in a nested dictionary.
+
+    In this context, a "magic value" is a string which is meant to be replaced by a different value
+    before the dictionary is accessed.
+
+    Args:
+        params: a (nested) dictionary of keys and values, where some values might be "magic"
+        magics: a dictionary of "magic values" (keys) and "real values" (values)
+
+    Returns: a (nested) dictionary where "magic values" are replaced by the "real values"
+
+    Examples:
+
+        >>> params_0 = {"key": "%magic_number%"}
```

Comment:
I'm wondering: What is the downside to passing a callable instead?

Reply:
let's make a new issue for that #189